resultify.py

```diff
@@ -46,6 +46,29 @@ def filter_emission(df: pd.DataFrame, emission: List) -> pd.DataFrame:
 
     return df[mask]
 
+def filter_primary_energy(df: pd.DataFrame, technologies: List) -> pd.DataFrame:
+    """Return rows that indicate Primary Energy use/generation
+    """
+    for t in range(len(technologies)):
+        technologies[t] = technologies[t].replace('*','')
+    df['REGION'] = df['TECHNOLOGY'].str[:2]
+    df_f = pd.DataFrame(columns=['REGION','TECHNOLOGY','FUEL','YEAR','VALUE'])
+    for t in range(len(technologies)):
+        if len(technologies[t])==2:
+            mask = df['TECHNOLOGY'].str[2:4] == technologies[t]
```

Comment:
Could you extract the position of the characters you match (`2:4`) from the `**` mask you pass in?  You wouldn't need the `if, else` logic then either.

Reply:
Not sure, how to do so. The issue is that I filter to [2:4] when to account the energy from Solar, Wind and Ocean, but to letter [4:9] (could also be only letter 6) when accounting for oil, gas and coal.

Reply:
One idea is to use [Regular Expressions](https://en.wikipedia.org/wiki/Regular_expression) in the configuration file, and then [pandas.str.contains](https://pandas.pydata.org/docs/reference/api/pandas.Series.str.contains.html).

You can try out reg ex patterns [here](https://regexr.com/)

Reply:
I gave it a go. Might be a bit more sophisticated to modify the config file. But the code is definitely simpler.